app/main/views/drafts.py

```diff
@@ -311,6 +314,31 @@ def delete_draft_service(draft_id):
         DraftService.id == draft_id
     ).first_or_404()
 
+    # Reset the source service - there should be a maximum of 1 copy per source service, as once
+    # a service is copied, it's removed from the list of available services to copy from.
+    # Resetting the .copied_to_following_framework flag on the source when the copy is deleted allows
+    # the user to try again with a fresh copy.
+    # TODO: remove this when draft.data['copiedFromServiceId'] is converted to a foreign key field
+    source_service, source_service_audit = None, None
+    if 'copiedFromServiceId' in draft.data:
+        source_service_id = draft.data['copiedFromServiceId']
+        source_service = Service.query.filter(
+            Service.service_id == source_service_id,
+            Service.copied_to_following_framework == True  # noqa
+        ).first()
+        if source_service:
+            source_service.copied_to_following_framework = False
+            source_service_audit = AuditEvent(
+                audit_type=AuditTypes.update_service,
+                user=updater_json['updated_by'],
+                data={
+                    "serviceId": draft.service_id,
+                    "supplierId": draft.supplier_id,
+                    "copiedToFollowingFramework": False
+                },
+                db_object=None
```

Comment:
I think you can still add it as the `db_object` - it's not actually enforced as a foreign key (can't be). Might need to add the "type" and "id" manually though so sqlalchemy doesn't try and be too clever and check its existence...

Reply:
Oh hold on I'm looking at the wrong audit event here - if it's the audit event for the service, it should _definitely_ have the object set shouldn't it?

Reply:
Good spot. Confession: I copied and pasted this from somewhere else. Will take a closer look.

Reply:
Yes it should. And it should get the data from the `source_service`, not the draft 🤦‍♀ Commit pushed to fix.